### DISCUSSION TITLE
Is it possible to bundle a single workspace of a monorepo?

### DISCUSSION THREAD

**Discussion by carlgieringer** [ojkelly/yarn.build#99] [Q&A] [answered]
Hi, does it work to bundle just one workspace of a monorepo?  I have a monorepo with multiple applications. An API server, a client, a browser extension. I would like to cd into the `api/` workspace subdirectory and run `yarn bundle` to bundle only that source and its dependencies. Is that supported?

When I try that now, I got a strange error relating to one of the other sub-folders. `infra` is a top-level directory in my monrepo, but it contains only terraform code. So I'm not sure what the error is. I would have expected yarn.build to ignore this directory.

```
Internal Error: ENOENT: no such file or directory, chmod '/private/var/folders/by/tbsr_6z13g37l2zld8xrg4hw0000gn/T/xfs-c8cbed0c/infra/.terraform/modules/295b6bd40c1b71e9361d2aebc0842ff5'
Error: ENOENT: no such file or directory, chmod '/private/var/folders/by/tbsr_6z13g37l2zld8xrg4hw0000gn/T/xfs-c8cbed0c/infra/.terraform/modules/295b6bd40c1b71e9361d2aebc0842ff5'
```

**Comment by ojkelly** [comment, marked as answer]
Ah, not yet. But it should be possible.

Yeah I would have thought it would have ignored that folder too, I have similar code like that.

Bundle is designed to take the target package, and it's dependencies (including local dependencies) and zip them all up. It then drops `entrypoint.js` which points to the `package.json#main` script, while loading `pnp.js`. Which primarily makes was lambda and containers work.

Provided the dir you want to bundle has a `package.json` inside it, and describes its dependencies it should work. Thats definitely the intent, so there might be a bug here.

Part of how it does this is by copying the repo to a temporary folder and deleting everything thats not required. Possibly something is calling your terraform code from the bundle? You could mark up `infra` as a package without any code by adding in a `package.json` and adding it as a dep to the other package that wants it.

**Comment by ojkelly** [comment]
I'm going to close this issue, but feel free to re-open or comment if you are still having trouble or have more information.

**Comment by bullno1** [comment]
I have the same problem.

I run `yarn bundle` inside the directory of a workspace.
`yarn bundle` complains about another workspace with `ENOENT` and that workspace only depends on the workspace being packed.

```
bullno1@alfheim ‹ master ● › : ~/Projects/mare/packages/jockey-cli
[1] % yarn bundle
Internal Error: ENOENT: no such file or directory, stat '/tmp/xfs-47e6b787/node_modules/mare-example'
    at Object.statSync (node:fs:1536:3)
    at Wt.statSync (/home/bullno1/Projects/mare/.yarn/releases/yarn-berry.cjs:317:2861)
    at Mt.removeEmptyDirectories (/home/bullno1/Projects/mare/.yarn/plugins/@ojkelly/plugin-all.cjs:44:3990)
    at Mt.removeEmptyDirectories (/home/bullno1/Projects/mare/.yarn/plugins/@ojkelly/plugin-all.cjs:44:4087)
    at async Mt.removeEmptyDirectories (/home/bullno1/Projects/mare/.yarn/plugins/@ojkelly/plugin-all.cjs:44:4076)
    at async Mt.removeExcluded (/home/bullno1/Projects/mare/.yarn/plugins/@ojkelly/plugin-all.cjs:44:4631)
    at async /home/bullno1/Projects/mare/.yarn/plugins/@ojkelly/plugin-all.cjs:45:1039
    at async Wt.mktempPromise (/home/bullno1/Projects/mare/.yarn/releases/yarn-berry.cjs:317:62926)
    at async Mt.execute (/home/bullno1/Projects/mare/.yarn/plugins/@ojkelly/plugin-all.cjs:44:4697)
    at async Mt.validateAndExecute (/home/bullno1/Projects/mare/.yarn/releases/yarn-berry.cjs:337:620)
```

**Reply by ojkelly** [reply]
Would you be able to provide a repo where this bug happens? If so can you open an issue for this, as  at the very least I’d love us to provide a better error message there.
